gcp_variant_transforms/bq_to_vcf.py

```diff
@@ -56,19 +65,78 @@ def run(argv=None):
   logging.info('Command: %s', ' '.join(argv or sys.argv))
   known_args, pipeline_args = vcf_to_bq_common.parse_args(argv,
                                                           _COMMAND_LINE_OPTIONS)
+  (project, dataset, table) = bigquery_util.parse_table_reference(
+      known_args.input_table)
   bq_source = bigquery.BigQuerySource(
       query=_BASE_QUERY_TEMPLATE.format(
-          INPUT_TABLE='.'.join(bigquery_util.parse_table_reference(
-              known_args.input_table))),
+          INPUT_TABLE='.'.join([project, dataset, table])),
       validate=True,
       use_standard_sql=True)
 
   options = pipeline_options.PipelineOptions(pipeline_args)
+
+  blob_prefix = '_sharded_vcf_files_{}'.format(
+      datetime.now().strftime('%Y%m%d_%H%M%S'))
+  file_path_prefix = '/'.join(['gs:/', known_args.bucket_name, blob_prefix])
   with beam.Pipeline(options=options) as p:
     _ = (p | 'ReadFromBigQuery ' >> beam.io.Read(bq_source)
          | bigquery_to_variant.BigQueryToVariant()
          | densify_variants.DensifyVariants()
-         | vcfio.WriteToVcf(known_args.output_file))
+         | 'PairVariantWithKey' >> beam.Map(_pair_variant_with_key,
+                                            _NUMBER_OF_VARIANTS_PER_SHARD)
+         | 'GroupVariantsByKey' >> beam.GroupByKey()
+         | vcfio.WriteVcfDataLines(file_path_prefix))
```

Comment:
I think the interface of WriteVcfDataLines class is confusing as it accepts a list and a key to use to generate the file name (it should really just get a list of variants and the full path of the file name). What do you think of introducing another simple transform after GroupByKey to generate the full file path, sort the variants, and then pass `(file_path, variants)` to a DoFn that actually writes the file?
Combined with my earlier suggestion of keeping the Coder class, the DoFn will just be a simple for loop over variants as it already has the full file path in it.